### ISSUE TITLE
document is not defined at Carousel:26

### ISSUE THREAD

**Issue by IAL32** [bestguy/sveltestrap#200]
I tried to just copy-paste the example noted [here](https://sveltestrap.js.org/?path=/story/components--carousel), the "With Captions" one, but I get this instead of the Carousel:

```
document is not defined

ReferenceError: document is not defined
    at [...]/node_modules/sveltestrap/src/Carousel.svelte:26:12
```

Which points to this:

```
  [...]
  onMount(() => {
    setRideTimeout();
  });

  _removeVisibilityChangeListener = browserEvent(
    document,     <------  this line!
    'visibilitychange',
    () => {
      if (document.visibilityState === 'hidden') {
        clearRideTimeout();
      } else {
        setRideTimeout();
      }
    }
  );
  [...]
```

According to [this StackOverflow answer](https://stackoverflow.com/a/56184892), the problem appears to be connected to the fact that `document` is only available after the component has been mounted.

I though, as a very noob on Svelte, to just chuck the assignment to `_removeVisibilityChangeListener` inside the `onMount` callback, and appears to be working just fine.
```
  onMount(() => {
    setRideTimeout();

    _removeVisibilityChangeListener = browserEvent(
      document,
      "visibilitychange",
      () => {
        if (document.visibilityState === "hidden") {
          clearRideTimeout();
        } else {
          setRideTimeout();
        }
      }
    );
  });
```

Node version: v12.19.0
Svelte: v3.17.3
Sveltestrap: v3.9.4
Sapper: v0.28.0

I will soon create a PR for this, hope this helps out anybody!

**Comment by bestguy** [comment]
Hi, are you. doing this in Sapper or server side rendering by chance?

**Comment by IAL32** [comment]
Yep, I have updated the issue with my current Sapper version: v0.28.0

**Comment by bestguy** [comment]
Thanks, and just to confirm, are you importing like mentioned here:
https://github.com/bestguy/sveltestrap#note-on-server-side-rendering-ssr-usage-fixed

and is sveltestrap in your dev dependencies?
(Just checking as there are always weird issues with Sapper SSR)

**Comment by IAL32** [comment]
Currently, my devDependencies are as follows:

```
  "devDependencies": {
    "@babel/core": "^7.0.0",
    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
    "@babel/plugin-transform-runtime": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "@babel/runtime": "^7.0.0",
    "@rollup/plugin-babel": "^5.0.0",
    "@rollup/plugin-commonjs": "^14.0.0",
    "@rollup/plugin-node-resolve": "^8.0.0",
    "@rollup/plugin-replace": "^2.2.0",
    "@rollup/plugin-url": "^5.0.0",
    "fa-svelte": "^3.1.0",
    "rollup": "^2.3.4",
    "rollup-plugin-svelte": "^6.0.0",
    "rollup-plugin-terser": "^7.0.0",
    "sapper": "^0.28.0",
    "svelte": "^3.17.3",
    "sveltestrap": "^3.9.4"
  }
```

And my copy-paste component is this:

```
<script>
  import {
    Carousel,
    CarouselControl,
    CarouselIndicators,
    CarouselItem,
    CarouselCaption
  } from "sveltestrap/src";

  const items = [
    {
      url:
        'data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22800%22%20height%3D%22400%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20800%20400%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_15ba800aa1d%20text%20%7B%20fill%3A%23555%3Bfont-weight%3Anormal%3Bfont-family%3AHelvetica%2C%20monospace%3Bfont-size%3A40pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_15ba800aa1d%22%3E%3Crect%20width%3D%22800%22%20height%3D%22400%22%20fill%3D%22%23777%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22285.921875%22%20y%3D%22218.3%22%3EFirst%20slide%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E',
      title: 'Slide 1',
      subTitle: 'Slide 1'
    },
    {
      url:
        'data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22800%22%20height%3D%22400%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20800%20400%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_15ba800aa20%20text%20%7B%20fill%3A%23444%3Bfont-weight%3Anormal%3Bfont-family%3AHelvetica%2C%20monospace%3Bfont-size%3A40pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_15ba800aa20%22%3E%3Crect%20width%3D%22800%22%20height%3D%22400%22%20fill%3D%22%23666%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22247.3203125%22%20y%3D%22218.3%22%3ESecond%20slide%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E',
      title: 'Slide 2',
      subTitle: 'Slide 2'
    },
    {
      url:
        'data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22800%22%20height%3D%22400%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20800%20400%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_15ba800aa21%20text%20%7B%20fill%3A%23333%3Bfont-weight%3Anormal%3Bfont-family%3AHelvetica%2C%20monospace%3Bfont-size%3A40pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_15ba800aa21%22%3E%3Crect%20width%3D%22800%22%20height%3D%22400%22%20fill%3D%22%23555%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22277%22%20y%3D%22218.3%22%3EThird%20slide%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E',
      title: 'Slide 3',
      subTitle: 'Slide 3'
    }
  ];
  let activeIndex = 0;
</script>

<Carousel {items} bind:activeIndex ride={true} interval={2000}>
  <CarouselIndicators bind:activeIndex {items} />

  <div class="carousel-inner">
    {#each items as item, index}
      <CarouselItem bind:activeIndex itemIndex={index}>
        <img src={item.url} class="d-block w-100" alt={item.title} />
        <CarouselCaption
          captionHeader={item.title}
          captionText={item.subTitle} />
      </CarouselItem>
    {/each}
  </div>

  <CarouselControl direction="prev" bind:activeIndex {items} />
  <CarouselControl direction="next" bind:activeIndex {items} />
</Carousel>
```

**Comment by bestguy** [comment]
Hi @IAL32 this is released in sveltestrap@3.9.5

Please let me know if looks good and welcome to close this.

**Comment by bestguy** [comment]
Will close, should be fixed in 3.9.5.  Please reopen if still seeing issues
